bodhi/server/metadata.py

```diff
@@ -306,3 +306,31 @@ def cache_repodata(self):
             shutil.rmtree(cache)
         shutil.copytree(repodata, cache)
         log.info('%s cached to %s' % (repodata, cache))
+
+
+class PungiMetadata(ExtendedMetadata):
+
+    def __init__(self, release, request, db, path, compose_dir):
+        super(PungiMetadata, self).__init__(release, request, db, path)
+        self.compose_dir = compose_dir
+
+    def modifyrepo(self, filename):
+        """Inject a file into the repodata for each architecture"""
+        dir_blacklist = ['source']
+        arches = [d for d in os.listdir(self.compose_dir) if d not in dir_blacklist]
+
+        for arch in arches:
+            repodata = os.path.join(self.compose_dir, arch, 'os', 'repodata')
+            log.info('Inserting %s into %s', filename, repodata)
+            uinfo_xml = os.path.join(repodata, 'updateinfo.xml')
+            shutil.copyfile(filename, uinfo_xml)
+            repomd_xml = os.path.join(repodata, 'repomd.xml')
+            repomd = cr.Repomd(repomd_xml)
+            uinfo_rec = cr.RepomdRecord('updateinfo', uinfo_xml)
+            uinfo_rec_comp = uinfo_rec.compress_and_fill(self.hash_type, self.comp_type)
+            uinfo_rec_comp.rename_file()
+            uinfo_rec_comp.type = 'updateinfo'
+            repomd.set_record(uinfo_rec_comp)
+            with file(repomd_xml, 'w') as repomd_file:
+                repomd_file.write(repomd.xml_dump())
+            os.unlink(uinfo_xml)
```

Comment:
This method is also almost identical to the superclass's version. Let's modify the superclass to make it flexible enough for them to share the code that's the same rather than duplicating so much code. That way they can both benefit from future bug fixes or enhancements.

Reply:
See https://github.com/fedora-infra/bodhi/pull/1876

Reply:
So i asked @puiterwijk on IRC today if i can look at failed tests on his PR https://github.com/fedora-infra/bodhi/pull/1876 if he does not have time for it and i wrote a patch https://github.com/fedora-infra/bodhi/pull/1891. So now all tests should pass.

Reply:
#1876 is merged, so let's go ahead and rebase this and fix it to use it.

Reply:
Don't forget to merge my PRs before rebasing or I will have sad.